playwright-e2e/pages/page-base.ts

```diff
@@ -0,0 +1,107 @@
+import { Locator, Page, Response } from '@playwright/test';
+import { PageInterface } from './page-interface';
+
+export default abstract class PageBase implements PageInterface {
+  protected readonly page: Page;
+  protected readonly baseUrl: string;
+
+  protected constructor(page: Page, baseURL: string) {
+    this.page = page;
+    this.baseUrl = baseURL;
+  }
+
+  abstract waitForReady(): Promise<void>;
+  abstract getBackButton(): Locator;
+
+  /**
+   * Returns "Log In" button locator
+   */
+  getLogInButton(): Locator {
+    return this.page.locator('.header button[data-ddp-test="signInButton"]:has-text("Log In")');
+  }
+
+  /**
+   * Returns "Log Out" button locator
+   */
+  getLogOutButton(): Locator {
+    return this.page.locator('.header button[data-ddp-test="signOutButton"]:has-text("Log Out")');
+  }
+
+  /**
+   * Return "Next" button locator
+   */
+  getNextButton(): Locator {
+    return this.page.locator('button', { hasText: 'Next' });
+  }
+
+  /**
+   * Returns "Submit" button locator
+   */
+  getSubmitButton(): Locator {
+    return this.page.locator('button', { hasText: 'Submit' });
+  }
+
+  /**
+   * Returns "I Agree" button locator
+   */
+  getIAgreeButton(): Locator {
+    return this.page.locator('button', { hasText: 'I agree' });
+  }
+
+  /**
+   * Returns "I'm not ready to agree" button
+   */
+  getIAmNotReadyToAgreeButton(): Locator {
+    return this.page.locator('button', { hasText: 'I am not ready to agree' });
+  }
+
+  async gotoURL(url: string): Promise<Response | null> {
+    return this.page.goto(url, { waitUntil: 'domcontentloaded' });
+  }
+
+  async gotoURLPath(urlPath = ''): Promise<Response | null> {
+    if (urlPath.startsWith('https')) {
+      throw Error('Parameter urlPath is not valid.');
+    }
+    return this.page.goto(`${this.baseUrl}${urlPath}`, { waitUntil: 'domcontentloaded' });
+  }
+
+  protected async clickAndWaitForNav(locator: Locator, opts: { waitForNav?: boolean } = {}): Promise<void> {
+    const { waitForNav = false } = opts;
+    waitForNav ? await this.waitForNavAfter(() => locator.click()) : await locator.click();
```

Comment:
It can be just 
`opts.waitForNav ? await this.waitForNavAfter(() => locator.click()) : await locator.click();`
without line 70.

Reply:
Just saw this now. I have merged PR already before seeing this. I don't have a strong opinion. Can you update this in your PR when rebasing? Sorry for not seeing your comment earlier.